cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/sync/class-storage.php

```diff
@@ -0,0 +1,242 @@
+<?php
+/**
+ * Storage management options.
+ *
+ * @package Cloudinary
+ */
+
+namespace Cloudinary;
+
+namespace Cloudinary\Sync;
+
+use Cloudinary\Component\Notice;
+use Cloudinary\Sync;
+
+/**
+ * Class Filter.
+ *
+ * Handles filtering of HTML content.
+ */
+class Storage implements Notice {
+
+	/**
+	 * Holds the Plugin instance.
+	 *
+	 * @since   0.1
+	 *
+	 * @var     \Cloudinary\Plugin Instance of the plugin.
+	 */
+	protected $plugin;
+
+
+	/**
+	 * Holds the Plugin Media instance.
+	 *
+	 * @since   0.1
+	 *
+	 * @var     \Cloudinary\Media Instance of the media object.
+	 */
+	protected $media;
+
+	/**
+	 * Holds the Sync instance.
+	 *
+	 * @since   0.1
+	 *
+	 * @var     \Cloudinary\Sync Instance of the plugin.
+	 */
+	protected $sync;
+
+	/**
+	 * Holds the Download Sync instance.
+	 *
+	 * @since   0.1
+	 *
+	 * @var     \Cloudinary\Sync\Download_Sync Instance of the plugin.
+	 */
+	protected $download;
+
+	/**
+	 * Holds the Connect instance.
+	 *
+	 * @since   0.1
+	 *
+	 * @var     \Cloudinary\Connect Instance of the plugin.
+	 */
+	protected $connect;
+
+	/**
+	 * Holds an array of the storage settings.
+	 *
+	 * @var array
+	 */
+	protected $settings;
+
+	/**
+	 * Filter constructor.
+	 *
+	 * @param \Cloudinary\Plugin $plugin The plugin.
+	 */
+	public function __construct( \Cloudinary\Plugin $plugin ) {
+		$this->plugin   = $plugin;
+		$this->sync     = $this->plugin->components['sync'];
+		$this->media    = $this->plugin->components['media'];
+		$this->download = $this->sync->managers['download'];
+
+		add_action( 'cloudinary_register_sync_types', array( $this, 'setup' ), 20 );
+	}
+
+	/**
+	 * Generate a signature for this sync type.s
+	 *
+	 * @return string
+	 */
+	public function generate() {
```

Comment:
Maybe this should be renamed to `generate_signature` for clarity

Reply:
Not a bad suggestion.